### PR TITLE
Feature/kenneth salt/renderhtml

### PR DESCRIPTION
**Pull request by kennethblite**
rendermup.js has been replaced by render_mup.js, running it will produce the desired files, added webstub.html that allows inlining of all dependencies.

### CODE REVIEW

File: scripts/checks.py [hunk 1]
```diff
@@ -234,7 +238,21 @@ def remove_hidden(node):
...
 	return
 
+if args.mupin is None:
+        fd_out =sys.stdout
+else:
+        fd_in.close()
+        fd_out=open(args.mupin, 'w')
+
 apply_each_node(root_node, remove_hidden)
+normalize_nodes(root_node)
```
Comment (BenGardiner) [inline scripts/checks.py]: looks like your checks.py changes crept in to this MR.

File: scripts/webstub.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<style>%%mupstyle%%</style>
+<svg width="2700" height="1060"></svg>
+<div id="container"><div id="chart"></div></div>
+<body>
+<script >%%d3script%%</script>
+<script >%%mupcommon%%</script>
+<script type="application/json" id="mupdata">
+%%jsondata%%
```
Comment (BenGardiner) [inline scripts/webstub.html]: ok awesome!

Is there a command-line to do the substitution into this file?

**Review by BenGardiner — CHANGES_REQUESTED** [review]

**Review by BenGardiner — APPROVED** [review]

**Comment by kennethblite** [comment]
@BenGardiner Do you want me to merge?

**Comment by BenGardiner** [comment]
@kennethblite 

This branch now contains everything in #15 also. I've tweaked your render_mup.js. It just does HTML output, rendermup (old) still does png files.

I'd like to open an issue to track the 'fork' between the two scripts. But otherwise this is ready to merge. And thank you again. The work you did here to get us over the hump of creating standalone HTML files has greatly improved the browsing experience of the trees.

**Pull request merged**